modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/reducers/fieldSetReducer.js

```diff
@@ -131,6 +134,29 @@ export default (state, action, config) => {
 				);
 			}
 
+			if (fieldName) {
+				return HandleSectionAdded(
+					props,
+					{
+						...state,
+						pages,
+					},
+					{
+						data: {
+							fieldName,
+							parentFieldName,
+						},
+						indexes,
+						newField: SettingsContext.updateField(
+							props,
+							fieldSetField,
+							'label',
+							fieldSet.name
+						),
+					}
+				);
+			}
+
```

Comment:
Hi @danvb112, I believe there is a code snippet that can be replaced by an existing function. Could you check if it's possible? The possibly replaceable code snippet is from line 91 to 176 of the `fieldsetReducer` by the `handleFieldSetAdded` function in `dynamic-data-mapping-form-builder`.

Reply:
Hi @miltonmc, I talked to @marcelabc about using the handler I mentioned in the comment above. She suggested that maybe this handler was deprecated, do you know anything about that?

Reply:
It's seen that most of  the *Reducers classes is a "copy" of *Handler classes. I imagine that, during metal war, the architecture has been reorganized, and some logic was copied to other classes (handler -> reducers). 
This ticket is a regression, but the original fix is in the Handler class but right now this code is not achievable. We have noticed that the same logic is in *Reducer but the fix is missing. We just copied the missing part from one class to another.

Reply:
@marcelabc you are right! Reducer at some point of the metal war copied all the logic into the old handlers, it was done to properly manage the form state and dispatch actions. Each reducer is a pluggable piece of the `FormProvider` allowing customizations on its use (using the strictly necessary and and new one for new use cases). The fixes you mentioned probably happened after the copy on some handler, but before the release of the new dispatch architecture, so they worked in this period but now they were missed.

Regarding  the  `dynamic-data-mapping-form-builder` it is legacy code and I'm working right now to remove all unused code and properly accommodate the rest of the code into other modules. Most of them will migrate to the  `data-engine/data-engine-js-components-web`. After that I think no handler will be available anymore!

Reply:
Thanks for the explanation, @miltommc! Will go resolve the conversation and aprove the PR.